### PR TITLE
Collect and enrich confluence by spaces

### PR DESCRIPTION
**Pull request by zhquan**
This code allows to collect and enrich confluence by spaces
adding `--spaces=[SPACE1, SPACE2]` on the projects.json.

For instance, the following projecs.json it will collect and
enrich the space `DEV` for the `Main` project and `SUPP`
space for `Support` project.

```
{
    "Main": {
        "confluence": [
            "https://example.com --spaces=[DEV]"
        ]
    },
    "Support": {
        "confluence": [
            "https://example.com --spaces=[SUPP]"
        ]
    }
}
```

Method updated:
- elastic_items.extract_repo_tags:
    - `tag`: New argument to select labels or spaces
- raw.elastc.get_perceval_params_from_url:
    - Get parameter `spaces` given a URL
- enriched.utils.get_confluence_spaces_filter:
    - Create a query to get items from `spaces`

Test added accordingly.

Signed-off-by: Quan Zhou <quan@bitergia.com>

### CODE REVIEW

File: releases/unreleased/collect-and-enrich-confluence-by-spaces.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+notes: >
+    The current version we have to collect the entire confluence
+    instance and add `--filter-raw` to enrich a specific space.
+
+    With this change we can collect and enrich only the specific
+    spaces adding `--spaces=[SPACE1, SPACE2, ...]` on the
+    projects.json.
+
+    Here and example of the projects.json to collect and enrich
+    only the `DEV` and `SUPP` spaces.
```
Comment (sduenas) [inline releases/unreleased/collect-and-enrich-confluence-by-spaces.yml:15]:
```suggestion
    Users can collect and enrich Confluence data from specific spaces only,
    adding the option `--spaces=[SPACE1, SPACE2, ...]` to their `projects.json`.

    The following example shows how to collect and to enrich data from
    only the `DEV` and `SUPP` spaces only.
```

File: tests/test_elastic_items.py [hunk 2]
```diff
@@ -119,6 +121,42 @@ def test_get_repository_filter_raw(self):
...
+        expected_filter = {}
+        perceval_backend_confluence = Confluence('https://example.com')
+        eitems = ConfluenceOcean(perceval_backend_confluence)
+        fltr = eitems.get_confluence_spaces(perceval_backend_confluence.spaces)
+        self.assertDictEqual(fltr, expected_filter)
+
+        # Meetup backend
+        expected_filter = {}
+        perceval_backend_meetup = Meetup('mygroup', 'aaaa')
+        eitems = MeetupOcean(perceval_backend_meetup)
```
Comment (sduenas) [inline tests/test_elastic_items.py:156]: I think this is a mistake and it should be there.

File: tests/test_elastic_items.py [hunk 3]
```diff
@@ -119,6 +121,42 @@ def test_get_repository_filter_raw(self):
...
+        fltr = eitems.get_confluence_spaces(['TEST'])
+        self.assertDictEqual(fltr, expected_filter)
```
Comment (sduenas) [inline tests/test_elastic_items.py:158]: And probably this should go replacing the other previous lines.

File: tests/test_elastic_items.py [hunk 4]
```diff
@@ -151,15 +189,38 @@ def test_set_repo_labels(self):
...
+        processed_repo, space_lst = eitems.extract_repo_tags("http://example.com --spaces=[A, B, C]", tag="spaces")
+
+        self.assertEqual(processed_repo, "http://example.com")
+        self.assertListEqual(space_lst, ['A', 'B', 'C'])
+
+        # spaces without tag='spaces'
+        processed_repo, space_lst = eitems.extract_repo_tags("http://example.com --spaces=[A, B, C]")
+
+        self.assertEqual(processed_repo, "http://example.com --spaces=[A, B, C]")
+        self.assertListEqual(space_lst, [])
```
Comment (sduenas) [inline tests/test_elastic_items.py:222]: Shouldn't go this in a different test? The name of the test is `test_extract_repo_labels`. Or maybe it makes sense to change the name.

**Review by sduenas — CHANGES_REQUESTED** [review]
I think the PR is ok but please check my comments.

File: tests/test_elastic_items.py (hunk 4 above)
Comment (zhquan) [reply tests/test_elastic_items.py:222]: Correct the name of the tests should be `test_extract_repo_tags`

File: tests/test_elastic_items.py (hunk 2 above)
Comment (zhquan) [reply tests/test_elastic_items.py:156]: I add Meetup there to check the result when the backend is not Confluence. In the next line I run `eitems.get_confluence_spaces`. I going to change the comment to make it clearer.

**Review by sduenas — APPROVED** [review]
LGTM

**Comment by sduenas** [comment]
Apparently the commits of this PR were merged in master but the PR wasn't updated.

**Comment by sduenas** [comment]
Let's close this because the code is already merged.

**Pull request closed without merging**